src/main/java/com/twcable/jackalope/impl/jcr/NodeTypeManagerImpl.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twcable.jackalope.impl.jcr;
+
+import javax.jcr.RepositoryException;
+import javax.jcr.UnsupportedRepositoryOperationException;
+import javax.jcr.nodetype.InvalidNodeTypeDefinitionException;
+import javax.jcr.nodetype.NoSuchNodeTypeException;
+import javax.jcr.nodetype.NodeDefinitionTemplate;
+import javax.jcr.nodetype.NodeType;
+import javax.jcr.nodetype.NodeTypeDefinition;
+import javax.jcr.nodetype.NodeTypeExistsException;
+import javax.jcr.nodetype.NodeTypeIterator;
+import javax.jcr.nodetype.NodeTypeManager;
+import javax.jcr.nodetype.NodeTypeTemplate;
+import javax.jcr.nodetype.PropertyDefinitionTemplate;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Simple implementation of {@link NodeTypeManager}. Use {@link #registerNodeType(NodeType)} to add NodeTypes.
+ */
+@SuppressWarnings("DuplicateThrows")
+public class NodeTypeManagerImpl implements NodeTypeManager {
+    private Map<String, NodeType> nameToNodeType = new LinkedHashMap<>();
```

Comment:
I think I'm missing the reason for a LinkedHashMap here?  Are you just guaranteeing that getPrimaryNodeTypes and getMixinNodeTypes always returns the nodetypes in the same order?